hypothesis-python/src/hypothesis/_strategies.py

```diff
@@ -1430,7 +1436,15 @@ def as_strategy(strat_or_callable, thing):
             "using register_type_strategy" % (thing,)
         )
     # Finally, try to build an instance by calling the type object
-    return builds(thing)
+    if not isabstract(thing):
+        return builds(thing)
+    subclasses = thing.__subclasses__()
+    if not subclasses:
+        raise ResolutionFailed(
+            "Could not resolve %r to a strategy, because it is an abstract type "
```

Comment:
This error message seems to suggest that the user should register a strategy for the abc `thing`, when they actually need to create a subclass of `thing` and register *that*.

Reply:
No, they really can register a strategy for the abstract type `thing` directly!

A related point is that while `all(issubclass(t, thing) for t in thing.__subclasses__())`, there may be other types which are considered to be subclasses of `thing` because of the `.register` method. Such registered types would be a natural thing to register for an abstract type with no subclasses-by-inheritance.